CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+Desk](https://jira.niis.org/servicedesk/customer/portal/1) using the
+`Report a software problem` request type.
+[Sign up](https://jira.niis.org/secure/Signup!default.jspa) for an account and
+get access to the [X-Road Service Desk](https://jira.niis.org/servicedesk/customer/portal/1).
+
+Another alternative to report security issues and vulnerabilities is the X-Road bug bounty program that is run on the Intigrity platform. [Visit the program details](https://app.intigriti.com/programs/niis/x-road/detail) to get started.
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to X-Road? You can start by looking through these `beginner` and `help-wanted` issues:
+
+- [Beginner issues](https://jira.niis.org/issues/?jql=project%20%3D%20XRDDEV%20AND%20status%20%3D%20%22To%20Do%22%20AND%20labels%20%3D%20beginner%20ORDER%20BY%20key%20DESC) - issues which should only require a few lines of code, and a test or two.
+- [Help wanted issues](https://jira.niis.org/issues/?jql=project%20%3D%20XRDDEV%20AND%20status%20%3D%20%22To%20Do%22%20AND%20labels%20%3D%20help-wanted%20ORDER%20BY%20key%20DESC) - issues which should be a bit more involved than beginner issues.
+
+X-Road can be developed locally. For instructions on how to do this, see the [build instructions](src/BUILD.md). Also,
+completing the X-Road Academy [Core Developer training](https://academy.x-road.global/courses/x-road-core-developer) is
+strongly recommended.
+
+### Submitting a Pull Request
+
+To ease the review work and to make clearer what changes are done, a pull request should contain one feature or bug fix. The larger the pull request is, the more complex it is to review. Pull requests are generally reviewed and accepted on first-come, first-served (FCFS) basis.
+
+If a pull request implements a new feature or a bigger change in an existing feature, it's strongly recommended to submit an enhancement request in advance and indicate in the enhancement request that the implementation will be provided too. In that way, it's possible to ensure in advance that the pull request will be approved.
+
+Also, in case you're planning to implement an existing backlog item, let NIIS know about your plans in advance to avoid duplicate work.
+
+In case a pull request includes changes in the architecture, it's strongly recommended to discuss the changes with NIIS in advance. Agreeing on the details of the changes upfront will likely speed up the approval process.
+
+#### Pull Request Checklist
+
+Before opening a pull request, it's recommended to review the following checklist and make sure that all the requirements are met.
+
+- Are the features OK to be accepted to the project?
+- Does the PR meet the style guides?
+- Is the source code available?
+- Are all the required dependencies available?
+- No merge conflicts?
+- Is there enough test coverage? The test coverage should be equal or higher than in the previous version.
```

Comment:
I rephrased this and clarified that the details are checked during the pull request review process. I think it's good to mention it in the checklist so that the contributor is aware of it.